xonsh/procs/specs.py

```diff
@@ -888,6 +883,23 @@ def run_subproc(cmds, captured=False, envs=None):
             print(f"TRACE SUBPROC: {cmds}, captured={captured}", file=sys.stderr)
 
     specs = cmds_to_specs(cmds, captured=captured, envs=envs)
+    if _should_set_title():
+        # context manager updates the command information that gets
+        # accessed by _current_job() when setting the terminal's title
+        with xpj.update_current_cmds(cmds):
+            # clear prompt level cache
+            XSH.env["PROMPT_FIELDS"].reset()
+            # The terminal's title needs to be set before starting the
```

Comment:
Does this means fields will be cleared multiple times between the pipeline?

this seems elaborate for setting a single field current-job. We can directly update that particular value?

Reply:
Yeah, that's a good idea. I've added a `reset_key()` method to `PromptFields` so that only `current_job` gets removed from the cache.